state/protocol/badger/snapshot.go

```diff
@@ -163,7 +163,9 @@ func (s *Snapshot) Identities(selector flow.IdentityFilter) (flow.IdentityList,
 
 	// get identities from the current epoch first
 	identities := setup.Participants.Copy()
-	lookup := identities.Lookup()
+
+	// sort the identities so the 'Exists' binary search works
+	identities = identities.Sort(order.ByNodeIDAsc)
```

Comment:
I think we can completely avoid the `Copy` here, as `Sort` already copies to not modify the original `IdentityList`
```golang
identities := setup.Participants.Sort(order.ByNodeIDAsc) // Sort copies internally 
```

Reply:
will do

Reply:
done